test/test_utils/ec2.py

```diff
@@ -904,3 +909,43 @@ def post_process_mxnet_ec2_performance(connection, log_location):
         return {"Throughput": total / n}
     else:
         raise ValueError("total: {}; n: {} -- something went wrong".format(total, n))
+
+
+@retry(reraise=True, wait=wait_fixed(60), stop=stop_after_attempt(5))
+def install_python_in_instance(context, python_version="3.9"):
+    """
+    Install sagemaker local test dependencies
+    :param context: Invoke Context / Fabric Connection object
+    :param python_version: str python version to install, such as 3.8, 3.9, etc.
+    :return: None
+    """
+    context.run("""ls ~/.pyenv || git clone https://github.com/pyenv/pyenv.git ~/.pyenv""")
+    # Need to configure PATH and PYENV_ROOT changes in alternative location because ~/.bashrc and ~/.profile are
+    # not used when using bash through Fabric Connection.
+    context.run("sudo chmod 666 /etc/profile.d/dlami.sh")
+    context.run("""echo 'export PYENV_ROOT="$HOME/.pyenv"' >> /etc/profile.d/dlami.sh""")
+    context.run(
+        """echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> /etc/profile.d/dlami.sh"""
+    )
+    context.run("""echo 'eval "$(pyenv init -)"' >> /etc/profile.d/dlami.sh""")
+    context.run("sudo chmod 644 /etc/profile.d/dlami.sh")
+
+    context.run("sudo apt-get update")
+    context.run(
```

Comment:
why are we explicitly installing additional packages here? will this be consistently needed across python versions?

Reply:
I've just written the installation steps from the pyenv installation instructions as code here. These are all pyenv dependencies used to install any python version. As of now, Ubuntu does not have a deb package to install pyenv. Therefore we need to clone the repo, install its dependencies, and execute the `bin/pyenv` script to install python at the required version from source.

Haven't validated each of the additional packages to see whether they're already installed in the AMI (which could change over time anyway).